staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -210,6 +210,7 @@ func getServerIdentificationPartialConfig(configAuthInfo clientcmdapi.AuthInfo,
 	configClientConfig.CAFile = configClusterInfo.CertificateAuthority
 	configClientConfig.CAData = configClusterInfo.CertificateAuthorityData
 	configClientConfig.Insecure = configClusterInfo.InsecureSkipTLSVerify
+	configClientConfig.ServerName = configClusterInfo.TLSServerName
```

Comment:
Is this really the only place in this package that needs to change? E.g., what about getCluster() elsewhere in the file?

Reply:
Yes, couldn't find any other place I need to modify.
(and e2e tests didn't show any failures ...)

`getCluster` seems to just reset the CA data when insecure flag is set.